app/assets/javascripts/app/reservation_time_field_adjustor.js.coffee

```diff
@@ -78,12 +89,24 @@ class window.ReservationTimeFieldAdjustor
           .addMinutes(@durationField().val()))
 
   _reserveStartChangeCallback: =>
+    # Wait until all the fields are filled before we do anything here
+    return unless @reserveStart.valid()
+
     duration = @durationField().val()
-    # Changing the start time will leave the duration alone, but change the
-    # end time to X minutes after the start time
-    endTime = @reserveStart.getDateTime().addMinutes(duration)
+    if duration
+      # Changing the start time will leave the duration alone, but change the
+      # end time to X minutes after the start time
+      endTime = @reserveStart.getDateTime().addMinutes(duration)
+      @reserveEnd.setDateTime(endTime)
+    else
+      # If there was not a duration already, calculate from start/end
+      if @calculateDuration() >= 0
+        @durationField().val(@calculateDuration())
+        @durationField().trigger("change")
+      else
+        # When the input start time was after than the end time, set the end time to the beginning
```

Comment:
Aren't you a little young to remember that song? Of course, I had repressed memories of that movie.